### DISCUSSION TITLE
Post adapter deprecation notices in more places

### DISCUSSION THREAD

**Discussion by olevski** [keycloak/keycloak#14233] [Adapter deprecation]
I am not sure how it is for the other adapters, but for the node-js adapter there are only 2 places that mention the deprecation that are very easy to miss:
- https://www.keycloak.org/2022/02/adapter-deprecation
- https://www.keycloak.org/archive/downloads-19.0.1.html

I think that mentioning the deprecation is warranted in at least the following locations:
- github repo: https://github.com/keycloak/keycloak-nodejs-connect
- npm: https://www.npmjs.com/package/keycloak-connect
- official docs (in multiple places):
  - https://www.keycloak.org/docs/latest/securing_apps/index.html#node-js-server-side
  - https://www.keycloak.org/docs/latest/securing_apps/#_nodejs_adapter

**Comment by nils-christian** [comment]
Absolutely agree. In my opinion the adapters shouldn't even be uploaded to Maven starting with 19.X. Otherwise a lot of developers might not even realize that the adapters are no longer supported.

**Reply by nils-christian** [reply]
I opened a ticket for this issue: https://github.com/keycloak/keycloak/issues/17341

**Comment by andrewmcguinness** [comment]
Note the latest live documentation still contains:

_Keycloak can be secured by supplied adapters that are usually easier to use and provide better integration with Keycloak._ 

(aside from the status of the supplied adapters, the statement is nonsense -- the goal is to secure the application, not to secure Keycloak)

https://www.keycloak.org/docs/latest/securing_apps/

The "Other OpenID Connect Libraries" should be moved from 2.5 to 2.2, and the existing 2.1-2.3 should be removed or pushed back and prominently labelled as legacy

**Comment by nikosmoum** [comment]
Can we do something more on this front please (mention the exact adapters that are deprecated)? At the very least, mark the specific adapter Java classes as deprecated in code.

**Comment by stianst** [comment]
Totally agree that we need to advertise this more widely, and we will work on that. Based on feedback we have revised our plans, please take a look at https://www.keycloak.org/2023/03/adapter-deprecation-update.html.
